thriftbp/server_middlewares.go

```diff
@@ -380,3 +383,59 @@ func RecoverPanic(name string, next thrift.TProcessorFunction) thrift.TProcessor
 		},
 	}
 }
+
+// PrometheusMetrics tracks Prometheus metrics specific to the Thrift service.
+func PrometheusMetrics(service string) thrift.ProcessorMiddleware {
+	return func(method string, next thrift.TProcessorFunction) thrift.TProcessorFunction {
+		process := func(ctx context.Context, seqID int32, in, out thrift.TProtocol) (success bool, err thrift.TException) {
+			start := time.Now()
+			var successLabel, exceptionTypeLabel, baseplateStatusCode, baseplateStatus string
+			activeRequestLabels := prometheus.Labels{
+				"thrift_service":               service,
+				"thrift_method":                method,
+				"thrift_success":               successLabel,
+				"thrift_exception_type":        exceptionTypeLabel,
+				"thrift_baseplate_status":      baseplateStatus,
+				"thrift_baseplate_status_code": baseplateStatusCode,
+			}
+			activeRequests.With(activeRequestLabels).Inc()
+
+			defer func() {
+				successLabel := "true"
+				if !success {
+					successLabel = "false"
+				}
```

Comment:
this is actually not part of the definition of the success label, we can remove these 4 lines (treat it as true as long as err is nil).

either use `successLabel := strconv.FormatBool(suppressor.Suppress(err) == nil)` (if we still allow error suppressing), or just `successLabel := strconv.FormatBool(err == nil)`.

Reply:
@fishy How do I know if we still allow error suppressing?